src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -120,48 +123,55 @@ object ContainerSerializer {
 }
 
 object VolumeSerializer {
-  def toProto(volume: Volume): Protos.Volume = volume match {
-    case p: PersistentVolume =>
-      Protos.Volume.newBuilder()
-        .setContainerPath(p.containerPath)
-        .setPersistent(PersistentVolumeInfoSerializer.toProto(p.persistent))
-        .setMode(p.mode)
-        .build()
-
-    case e: ExternalVolume =>
-      Protos.Volume.newBuilder()
-        .setContainerPath(e.containerPath)
-        .setExternal(ExternalVolumeInfoSerializer.toProto(e.external))
-        .setMode(e.mode)
-        .build()
-
-    case d: DockerVolume =>
-      Protos.Volume.newBuilder()
-        .setContainerPath(d.containerPath)
-        .setHostPath(d.hostPath)
-        .setMode(d.mode)
-        .build()
-
-    case s: SecretVolume =>
-      Protos.Volume.newBuilder()
-        .setContainerPath(s.containerPath)
-        .setSecret(
-          Protos.Volume.SecretVolumeInfo.newBuilder().setSecret(s.secret).build()
-        )
-        .setMode(s.mode)
-        .build()
+  def toProto(volumeWithMount: VolumeWithMount): Protos.Volume = {
+    val volume = volumeWithMount.volume
+    val mount = volumeWithMount.mount
+    val mode = if (mount.readOnly) mesos.Protos.Volume.Mode.RO else mesos.Protos.Volume.Mode.RW
+
+    volume match {
+      case p: PersistentVolume =>
+        Protos.Volume.newBuilder()
+          .setContainerPath(mount.mountPath)
```

Comment:
I think I'd prefer to see the common fields extracted above the match statement.

IE:

```scala
  def toProto(volumeWithMount: VolumeWithMount): Protos.Volume = {
    val volume = volumeWithMount.volume
    val mount = volumeWithMount.mount
    val mode = if (mount.readOnly) mesos.Protos.Volume.Mode.RO else mesos.Protos.Volume.Mode.RW

    val volumeBuilder = Protos.Volume.newBuilder()
      .setContainerPath(mount.mountPath)
      .setMode(mode)

    volume match {
      case p: PersistentVolume =>
        volumeBuilder
          .setPersistent(PersistentVolumeInfoSerializer.toProto(p.persistent))

      case e: ExternalVolume =>
        Protos.Volume.newBuilder()
          .setExternal(ExternalVolumeInfoSerializer.toProto(e.external))

      case d: HostVolume =>
        Protos.Volume.newBuilder()
          .setHostPath(d.hostPath)

      case s: SecretVolume =>
        Protos.Volume.newBuilder()
          .setSecret(Protos.Volume.SecretVolumeInfo.newBuilder().setSecret(s.secret))
    }

    volumeBuilder.build()
  }

```

Reply:
Will do.